### PR TITLE
[Auth] Add gmpid header

### PR DESCRIPTION
**Pull request by sam-gc**

### CODE REVIEW

**Review by lisajian — APPROVED** [review]
LGTM - Is there an internal bug and/or issue associated with this change?

**Comment by sam-gc** [comment]
b/205577736

**Comment by lisajian** [comment]
Thanks!

File: packages/auth/src/core/auth/auth_impl.ts [hunk 1]
```diff
@@ -577,8 +578,15 @@ export class AuthImpl implements AuthInternal, _FirebaseService {
   _getFrameworks(): readonly string[] {
     return this.frameworks;
   }
-  _getSdkClientVersion(): string {
-    return this.clientVersion;
+  async _getAdditionalHeaders(): Promise<Record<string, string>> {
+    // Additional headers on every request
+    const headers: Record<string, string> = {
+      [HttpHeader.X_CLIENT_VERSION]: this.clientVersion,
```
Comment (yuchenshi) [inline packages/auth/src/core/auth/auth_impl.ts:584]: Is `this.clientVersion` always non-empty and non-null? If unsure, please add a check. Empty header values may break emulators)

**Review by yuchenshi — APPROVED** [review]

File: packages/auth/src/core/auth/auth_impl.ts (hunk 1 above)
Comment (sam-gc) [reply packages/auth/src/core/auth/auth_impl.ts:584]: This is just moved code (the prior revision also always sends it). It's also always populated

**Pull request merged**